apps/daimo-mobile/src/view/shared/useSwipeUpDown.tsx

```diff
@@ -0,0 +1,95 @@
+import { SCREEN_WIDTH } from "@gorhom/bottom-sheet";
+import { useIsFocused } from "@react-navigation/native";
+import { ReactNode, useEffect, useRef, useState } from "react";
+import { Dimensions, Platform, StyleSheet } from "react-native";
+import Animated, { useAnimatedStyle } from "react-native-reanimated";
+
+import { SwipeUpDown, SwipeUpDownRef } from "./SwipeUpDown";
+import { useNav } from "./nav";
+import useTabBarHeight from "../../common/useTabBarHeight";
+
+const screenDimensions = Dimensions.get("screen");
+
+export function useSwipeUpDown({
+  itemMini,
+  itemFull,
+  translationY,
+  disabled,
+}: {
+  itemMini: ReactNode;
+  itemFull: ReactNode;
+  translationY: Animated.SharedValue<number>;
+  disabled?: boolean;
+}) {
+  const [isBottomSheetOpen, setIsOpen] = useState(false);
+
+  // Dimensions
+  const tabBarHeight = useTabBarHeight();
+  const screenHeight =
+    screenDimensions.height -
+    tabBarHeight -
+    (Platform.OS === "android" ? 16 : 0);
+
+  // Hide bottom sheet when tapping a bottom tab.
+  const nav = useNav();
+  const bottomSheetRef = useRef<SwipeUpDownRef>(null);
+  const isFocused = useIsFocused();
+  useEffect(() => {
+    if (nav.getParent()) {
+      // @ts-ignore
+      const unsub = nav.getParent().addListener("tabPress", (e: Event) => {
```

Comment:
nit: could remove the `ts-ignore` here by checking / early exiting if `nav.getParent()` is null I think

Reply:
nah it also doesn't like `tabPress` / thinks that's not an event